tests/unit/StandaloneDirectorsFiling.spec.ts

```diff
@@ -1385,3 +1385,141 @@ describe('Standalone Directors Filing - Part 6 - Error/Warning Dialogs', () => {
+              annualGeneralMeetingDate: '2018-07-15'
+            },
+            business: {
+              cacheId: 1,
+              foundingDate: '2007-04-08',
+              identifier: 'CP0001191',
+              lastLedgerTimestamp: '2019-04-15T20:05:49.068272+00:00',
+              legalName: 'Legal Name - CP0001191'
+            },
+            header: {
+              name: 'annualReport',
+              date: '2017-06-06',
+              submitter: 'cp0001191',
+              status: 'DRAFT',
+              certifiedBy: 'Full Name',
+              email: 'no_one@never.get',
+              filingId: 123
+            }
+          }
+        }
+      }
+    })
+
+    p1.catch(() => {}) // pre-empt "unhandled promise rejection" warning
+
+    sinon
+      .stub(axios, 'post')
+      .withArgs('businesses/CP0001191/filings')
+      .returns(p1)
+  })
+
+  it('Attempt to file and pay with a BCOL error', async () => {
+    // set necessary session variables
+    sessionStorage.setItem('BASE_URL', `${process.env.VUE_APP_PATH}/`)
+    sessionStorage.setItem('PAY_API_URL', '')
+    sessionStorage.setItem('AUTH_URL', 'auth/')
+    const get = sinon.stub(axios, 'get')
+
+    get.withArgs('businesses/CP0001191/tasks')
+    .returns(new Promise(resolve => resolve({ data: { tasks: [] } })))
+
+    const $route = { params: { filingId: 0 } } // new filing id
+    wrapper = mount(StandaloneDirectorsFiling, {
+      store,
+      mocks: { $route },
+      stubs: {
+        CodDate: true,
+        Directors: true,
+        Certify: true,
+        StaffPayment: true,
+        Affix: true,
+        SbcFeeSummary: true,
+        ConfirmDialog: true,
+        PaymentErrorDialog: true,
+        ResumeErrorDialog: true,
+        SaveErrorDialog: true
+      },
+      vuetify
+    })
+
+    vm = wrapper.vm as any
+
+    // make sure form is validated
+    vm.inFilingReview = true
+    vm.codDateValid = true
+    vm.directorFormValid = true
+    vm.staffPaymentFormValid = true
+    vm.certifyFormValid = true
+    store.state.filingData = [{ filingTypeCode: 'OTCDR', entityType: 'CP' }] // dummy data
+
+    // stub address data
+    vm.addresses = {
+      registeredOffice: {
+        deliveryAddress: {},
+        mailingAddress: {}
+      }
+    }
+
+    // make sure a fee is required
+    vm.totalFee = 100
+
+    // sanity check
+
+    expect(vm.bcolErrMsg).toBeNull()
+    expect(vm.bcolTitle).toBeNull()
+
+    expect(vm.bcolErrorDialog).toBe(false)
+
+    const button = wrapper.find('#cod-file-pay-btn')
+    expect(button.attributes('disabled')).toBeUndefined()
+    
+    get.withArgs('codes/errors/BCOL_ERROR')
+    .returns(new Promise(resolve => resolve({ data: {
+      detail: 'An Error has occured',
+      title: 'Error'
+    }})))
+    // click the File & Pay button
+    await button.trigger('click')
+    await flushPromises()
+    await vm.onClickFilePay()
+    // work-around because click trigger isn't working
+
+    // verify redirection
```

Comment:
Update this comment.

Reply:
2 out of 3 were fixed. Please also fix line 1867 in StandaloneOfficeAddressFiling.spec.ts.

Reply:
Fixed.